athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/serde/v24/GetTableRequestSerDe.java

```diff
@@ -0,0 +1,83 @@
+/*-
+ * #%L
+ * Amazon Athena Query Federation SDK
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connector.lambda.serde.v24;
+
+import com.amazonaws.athena.connector.lambda.domain.TableName;
+import com.amazonaws.athena.connector.lambda.metadata.GetTableRequest;
+import com.amazonaws.athena.connector.lambda.request.FederationRequest;
+import com.amazonaws.athena.connector.lambda.security.FederatedIdentity;
+import com.amazonaws.athena.connector.lambda.serde.TypedSerDe;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+
+import java.io.IOException;
+
+import static java.util.Objects.requireNonNull;
+
+public class GetTableRequestSerDe
+        extends TypedSerDe<FederationRequest>
+{
+    private static final String IDENTITY_FIELD = "identity";
+    private static final String QUERY_ID_FIELD = "queryId";
+    private static final String CATALOG_NAME_FIELD = "catalogName";
```

Comment:
Thoughts on an abstract RequestSerDe so that you don't need these, and thus the code for these on every concrete request?